internal/jimm/bootstrap/bootstrap_job.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2025 Canonical.
+
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	jujucloud "github.com/juju/juju/cloud"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/jujuclient"
+	jujuparams "github.com/juju/juju/rpc/params"
+	"github.com/juju/zaputil/zapctx"
+	"go.uber.org/zap"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/jimm/juju"
+	"github.com/canonical/jimm/v3/internal/jobtracker"
+	"github.com/canonical/jimm/v3/internal/jujuclistore"
+	"github.com/canonical/jimm/v3/internal/jujucommands"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// BootstrapExecutor holds a wrapper to run a command. It is primarily for testing
+// and when running the bootstrap command the implemention to be used is [DefaultBootstrapExecutor].
+type BootstrapExecutor interface {
```

Comment:
interface should be defined where it is used, not where it is implemented

Reply:
I'm aware - but in this circumstance I had no other way to test this. I also asked the @kian99  on this one about using a function vs interface, and he was in favour of an interface.